### PR TITLE
Implemented Wired Hardware Sync for Azure Kinect Sensors.

### PR DESCRIPTION
**Pull request by xiangzhi**
Implementation of Request #76. This PR has the following changes:
1. Expose the parameters to enable hardware synchronization of multiple Azure Kinect sensors. It also included some checks to make sure the wired are connected and throw `ArgumentExceptions` otherwise.
2. Updated the `AzureKinectSample` to include an example of synchronization between multiple Azure Kinect Sensors. It also serves as a testing platform.
3. Added a lock when creating a camera instance as it sometimes fail when opening multiple Azure Kinects. 

I'm not too sure if it followed the correct style, so please let me know what needed to be changed!

### CODE REVIEW

**Comment by chitsaw** [comment]
@xiangzhi Thank you for submitting this PR and for your contribution to \psi! Fyi, as we review this, we are just about to push some updates today, so could we ask that you wait for those to hit the repo and merge from upstream prior to completing this PR? Thanks so much!

**Comment by xiangzhi** [comment]
> @xiangzhi Thank you for submitting this PR and for your contribution to \psi! Fyi, as we review this, we are just about to push some updates today, so could we ask that you wait for those to hit the repo and merge from upstream prior to completing this PR? Thanks so much!

No problem, I'll merge it with the latest push later.

**Comment by xiangzhi** [comment]
@chitsaw Just merged in the latest push. There wasn't any conflicts on the code level. Thanks!

File: Samples/AzureKinectSample/README.md
Comment (chitsaw) [inline Samples/AzureKinectSample/README.md:42]: The image capture delay between the primary and subordinate sensors.

File: Samples/AzureKinectSample/Program.cs
Comment (chitsaw) [reply Samples/AzureKinectSample/Program.cs:38]: Btw, our samples will be moving real soon to a separate repo to reduce make them more accessible. They will build against published/released NuGet packages.

File: Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs
Comment (chitsaw) [inline Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs:18]: Maybe there should be an additional `Default = 0` value here, which means  we don't call `this.device.SetColorControl` with an explicit `PowerlineFrequency` in `AzureKinectCore` and defer to whatever the core library defaults to.

File: Samples/AzureKinectSample/Program.cs
Comment (xiangzhi) [reply Samples/AzureKinectSample/Program.cs:38]: Cool, I'll drop the samples and update the wiki.

**Review by chitsaw — CHANGES_REQUESTED** [review]
Thanks again for submitting this PR - do let us know if you need help with anything to get these changes merged in.

File: Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs
Comment (xiangzhi) [reply Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs:17]: I named it `PowerlineFrequencyTypes` to prevent naming conflict with the variable name `PowerlineFrequency` used in the configuration. I don't know what is the best convention in this case.

File: Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs
Comment (xiangzhi) [reply Sources/Kinect/Microsoft.Psi.AzureKinect.x64/AzureKinectSensorConfiguration.cs:18]: I added a default and only sets the powerfrequency if its not the default. The Azure SDK only takes 1 or 2 and will throw an exception if zero is passed in.

**Comment by xiangzhi** [comment]
@chitsaw  Thanks for following up, I was swamp with other work stuff and just got around to work on this. Here's the updated branch with all the suggested changes. The only unresolved issue is the naming of the `enum`, please let me know what you think.

I also updated the wiki but I don't believe there is an easy way to merge them, so feel free to copy and paste them back here: https://github.com/xiangzhi/psi-wiki/blob/zhi-azure-kinect-updates/Azure-Kinect-Overview.md

**Review by chitsaw — APPROVED** [review]
Thanks for this PR. Looks good!

**Comment by chitsaw** [comment]
> I also updated the wiki but I don't believe there is an easy way to merge them, so feel free to copy and paste them back here: https://github.com/xiangzhi/psi-wiki/blob/zhi-azure-kinect-updates/Azure-Kinect-Overview.md

Yeah, I don't belive github supports forking and pulling for the wiki repo. I thought you would be able to just clone psi.wiki, branch and pull it in directly, but maybe there is a permissions issue that we'll have to look into. No bother, for now we'll just copy and paste once you complete your PR. Thanks again!

**Comment by xiangzhi** [comment]
No problem! Glad to help.

**Pull request merged**
